src/components/Join/Join.js

```diff
@@ -9,13 +9,13 @@ class Join extends Component {
     };
   }
 
-  handleChange = (e) => {
+  handleChange = (e) => { // eslint-disable-line
```

Comment:
This is false-positive, and it doesn't happen every time or on every machine, please check out the issue I mentioned in the PR description. This is a bug of eslint or babel-eslint and it should be fixed in the future.

As for workarounds, there are many of them, including binding them manually in constructor. Actually the method you gave looks less appealing than that, and it seems won't work. `handleSubmit` is still not bound to the class context.